### PR TITLE
Enhance Storybook

### PR DESCRIPTION
**Pull request by timoheddes**
This PR aims to enhance our Storybook by fixing bugs, adding features, adding documents and configuring addons.

**Updates/changes:**
- Added a Welcome section that loads first and shows our Readme as Intro and our Contribution guide as stories.
- Configured our Storybook theme to display the Kadena logo and use our preferred fonts.
- Attempts to split and tidy up Storybook's configuration files.
- Fixes the light/dark mode when in Docs view of stories (previously these would not apply the dark theme).
- Introduces story decorators.
- Updates various stories that were missing descriptions.
- Introduces new subgroups for components and attempts to organise our stories better.

[Related Asana task](https://app.asana.com/0/1205163122576860/1205221706151250/f)

https://github.com/kadena-community/kadena.js/assets/79908211/e6213f2b-a427-4912-befb-4b0df7dded12

### CODE REVIEW

File: packages/libs/react-ui/src/Intro.mdx [hunk 1]
```diff
@@ -0,0 +1,281 @@
+import { Meta } from '@storybook/blocks';
+
+<Meta
+  title="Welcome/Intro"
+  parameters={{ options: { isToolshown: false }, id: 1 }}
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/Intro.mdx:5]: The Intro and Contributing files are complete duplicates of existing md files. Can we just reference those instead and add this meta data there?

**Review by eileenmguo — COMMENTED** [review]
Really excited about this!

File: packages/libs/react-ui/src/Intro.mdx (hunk 1 above)
Comment (timoheddes) [reply packages/libs/react-ui/src/Intro.mdx:5]: Yeah true, that would make more sense. I'll have a look!

File: packages/libs/react-ui/src/components/Stack/Stack.stories.tsx [hunk 2]
```diff
@@ -14,6 +15,7 @@ const spaceOptions: (keyof typeof vars.sizes | undefined)[] = [
 
 const meta: Meta<typeof Stack> = {
   title: 'Layout/Stack',
+  decorators: [withCenteredStory],
```
[the remark this answers is not in the document]
Comment (timoheddes) [reply packages/libs/react-ui/src/components/Stack/Stack.stories.tsx:18]: You might not want it for every story though. This is just an example of decorators we could make and use. I've applied this one where it seemed to make sense.

File: packages/libs/react-ui/.storybook/preview.ts [hunk 3]
```diff
@@ -13,13 +16,29 @@ const preview: Preview = {
...
+      current: 'light',
       stylePreview: true,
-      darkClass: darkThemeClass,
+      classTarget: 'html',
+      darkClass: [darkThemeClass, 'dark-mode'],
+      lightClass: ['light-mode'],
       // Override the default dark theme
       dark: { ...themes.dark, appBg: 'black' },
       // Override the default light theme
       light: { ...themes.normal, appBg: 'white' },
```
[the remark this answers is not in the document]
Comment (timoheddes) [reply packages/libs/react-ui/.storybook/preview.ts:27]: I think this was already there but yeah sure! Makes sense

**Comment by timoheddes** [comment]
Removed draft status. Like I said I think we are continuously improving on Storybook and this already adds a lot. We can expand on it further in a later PR.

**Review by eileenmguo — APPROVED** [review]

**Pull request merged**